src/neural-network.js

```diff
@@ -366,6 +392,22 @@ export default class NeuralNetwork {
     return true;
   }
 
+
+  /**
+   *
+   * @param options
+   * @private
+   */
+  _validateTrainingOptions(options) {
+    Object.keys(options).forEach(key => {
```

Comment:
Why not iterate over the `trainingValidations` functions? Only care about things we care about?

Reply:
yeah I only wanted to check the things that actually are changing.  I could see it going either way (the cost isn't huge).